src/server/system_services/objects/quota.js

```diff
@@ -0,0 +1,198 @@
+         this.size[RAW_VALUE_FIELD_NAME] = size_utils.size_unit_to_bigint(this.size.value, this.size.unit).toJSON();
+         this.quantity = quota_config && quota_config.quantity ? quota_config.quantity : {'value': 0};
+         this.quantity[RAW_VALUE_FIELD_NAME] = new BigInteger(this.quantity.value).toJSON();
+    }
+
+    /**
+     * @returns size value.
+     */
+    get_size_value() {
+        return this.size.value;
+    }
+
+    /**
+     * @returns size unit.
+     */
+    get_size_unit() {
+        return this.size.unit;
+    }
+
+    /**
+     * @returns quantity value.
+     */
+    get_quantity_value() {
+        return this.quantity.value;
+    }
+
+    /**
+     * @returns quantity unit.
+     */
+     get_quantity_unit() {
+        return this.quantity.unit;
+    }
+
+    /**
+     * 
+     * @returns calculated bigint size raw value of quota. default is 0.
+     */
+    get_quota_by_size() {
+        return this.size.raw_value;
+    }
+
+    /**
+     * 
+     * @returns calculated bigint quantity raw value of quota
+     */
+    get_quota_by_quantity() {
+        return this.quantity.raw_value;
+    }
+
+    /**
+     * 
+     * @returns - new quota config object without raw values
+     */
+    get_config() {
+        const quota_config = {};
+        if (this.size.value > 0) {
+            quota_config.size = _.omit(this.size, RAW_VALUE_FIELD_NAME);
+        }
+        if (this.quantity.value > 0) {
+            quota_config.quantity = _.omit(this.quantity, RAW_VALUE_FIELD_NAME);
+        }
+        return quota_config;
+    }
+
+
+  /**
+   * 
+   * @returns the percent of quantity quota used by the bucket. 
+   */
+    static get_bucket_quota_usages_percent(bucket, bucket_quota) {
+        var usage = {
+            size_used_percent: 0,
+            quantity_used_percent: 0
+        };
+        if (!bucket_quota) return usage;
+        const objects_size = bucket.storage_stats && size_utils.json_to_bigint(bucket.storage_stats.objects_size);
+        const quota_size = size_utils.json_to_bigint(bucket_quota.get_quota_by_size());
+        if (quota_size > 0) {
+            usage.size_used_percent = objects_size.multiply(100).divide(quota_size).valueOf();
+        }
+        const objects_count = bucket.storage_stats && size_utils.json_to_bigint(bucket.storage_stats.objects_count);
+        const quantity_quota = size_utils.json_to_bigint(bucket_quota.get_quota_by_quantity());
+        if (quantity_quota > 0) {
+            usage.quantity_used_percent = objects_count.multiply(100).divide(quantity_quota).valueOf();
+        }
+        return usage;
+    }
+
+    /**
+     * Create and add quota alerts to collection if usage greates than threshold 90%.
+     * @param {*} sysid sysid - system id
+     * @param {*} bucket - own bucket
+     * @param {*} quota - bucket quota
+     * @returns
+     */
+    static add_quota_alerts(sysid, bucket, quota, collection) {
```

Comment:
I think it's better to decouple the alerts logic from the quota logic. maybe have a `check_quota` function that returns errors and then handle these errors in the object_server

Reply:
the method is used in object_server and quota_server.